pptx/text/bullets.py

```diff
@@ -0,0 +1,720 @@
+        """
+        bullet_color_elm = parent.eg_textBulletColor
+        bullet_color = _BulletColor(bullet_color_elm)
+        text_bullet_color = cls(parent, bullet_color)
+        return text_bullet_color
+
+    def follow_text(self):
+        """
+        Sets the BulletColor to _BulletColorFollowText
+        """
+        follow_text = self._parent.get_or_change_to_buClrTx()
+        self._bullet_color = _BulletColorFollowText(follow_text)
+
+    def set_color(self, color):
+        """
+        Sets the BulletColor to _BulletColorSpecific and sets the color
+        """
+        bullet_color = self._parent.get_or_change_to_buClr()
+        self._bullet_color = _BulletColorSpecific(bullet_color)
+        self.color = color
+
+    @property
+    def type(self):
+        """ Return a string type """
+        return self._bullet_color.type
+    
+    @property
+    def color(self):
+        """ Return the |ColorFormat| object """
+        return self._bullet_color.color
+    
+    @color.setter
+    def color(self, value):
+        """
+        Set the value of the color as either an RGB or Theme Color
+        """
+        if not isinstance(self._bullet_color, _BulletColorSpecific):
+            raise TypeError("BulletColor is not of type BulletColorSpecific")
+        color_obj = self._bullet_color.color
+        if isinstance(value, RGBColor):
+            color_obj.rgb = value
+        elif isinstance(value, _SchemeColor):
+            color_obj.theme_color = value
+        else:
+            raise TypeError("Provided color value is incorrect type")
+     
+
+class _BulletColor(object):
+    """
+    Object factory for BulletColor objects
+    """
+
+    def __new__(cls, xBulletColor):
+        if isinstance(xBulletColor, CT_TextBulletColorFollowText):
+            bullet_color_cls = _BulletColorFollowText
+        elif isinstance(xBulletColor, CT_Color):
+            bullet_color_cls = _BulletColorSpecific
+        else:
+            bullet_color_cls = _BulletColor
+        
+        return super(_BulletColor, cls).__new__(bullet_color_cls)
+        
+    @property
+    def type(self):
+        return "NoBulletColor"
+
+    @property
+    def color(self):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletColor type %s has no color, call .set_color() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @color.setter
+    def color(self, value):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletColor type %s has no color, call .set_color() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+
+
+class _BulletColorFollowText(_BulletColor):
+    """
+    Designates that the Bullet Color will match the accompanying paragraph text.
+    """
+    @property
+    def type(self):
+        return "BulletColorFollowText"
+
+class _BulletColorSpecific(_BulletColor):
```

Comment:
I'm not entirely sure what BulletColorSpecific means. Can you add a comment here to clarify?